hx_lti_initializer/static/vendors/development/video.dev.js

```diff
@@ -6309,6 +6320,10 @@ vjs.TextTrackMenuItem = vjs.MenuItem.extend({
 vjs.TextTrackMenuItem.prototype.onClick = function(){
   vjs.MenuItem.prototype.onClick.call(this);
   this.player_.showTextTrack(this.track.id_, this.track.kind());
+  if (typeof(jQuery.publish) === "function") {
```

Comment:
It looks like you're modifying the vendor source for VideoJS in at least 5 spots here in order to publish events about actions taken. We should probably document any/all changes to vendor libraries, because I could totally see us trying to upgrade VJS later and forgetting we made these changes.

As an alternative to directly modifying the functions themselves, have you considered decorating them in some way, which would separate out the customizations a bit from the vendor code? It might not be possible in some instances, but if you just need to know that a function is called, that might be just as effective.

Something like this perhaps:

 ```javascript
(function(vjs) {
 
  var onClickTextTrack = vjs.TextTrackMenuItem.prototype.onClick;
  vjs.TextTrackMenuItem.prototype.onClick = function() {
    onClickTextTrack.apply(this, arguments);
    jQuery.publish("captions_toggled", [this.track.label()]);
  };

})(window.videojs);
```

Not pretty, but at least if the vendor library needed to be updated, the custom code is separate and could be layered on top, or at least easily identified if anything broke. Just a thought for the future.